host/xtest/regression_1000.c

```diff
@@ -1381,24 +1386,58 @@ static void xtest_tee_test_1017(ADBG_Case_t *c)
 	TEEC_ReleaseSharedMemory(&shm);
 }
 
+static void set_outbound_by_1byte_shm_ref(TEEC_Operation *op,
+					  TEEC_SharedMemory *shm)
+{
+	memset(op, 0, sizeof(*op));
+
+	op->paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_PARTIAL_INPUT,
+					 TEEC_MEMREF_PARTIAL_INPUT,
+					 TEEC_MEMREF_PARTIAL_OUTPUT,
+					 TEEC_MEMREF_PARTIAL_OUTPUT);
+
+	/*
+	 * The first three memrefs are valid references while last is outbound
+	 * by 1 byte and should lead to failure to invoke the TA.
+	 */
+	op->params[0].memref.parent = shm;
```

Comment:
Why do we have these three valid memrefs first? To me it looks like it only muddies the water.

Reply:
The TA expects 4 params so we have to provide 4.
Yet, we could use 4 invalid references...
I made it to slightly match the designed of the original case 1018.

Reply:
Please add another TAF or modify it to accept only one parameter too. Especially when it comes to negative tests it's good to not do more than necessary to avoid that something else hides the error.